symfony/src/Service/SiteSyncService.php

```diff
@@ -0,0 +1,257 @@
+                        $siteData->setWorkqueueDownload('full_data'); // default value for workqueue downlaod
+                    }
+                    if ($existing) {
+                        if ($existing != $siteData) {
+                            $modified[] = [
+                                'old' => $existing->toArray(),
+                                'new' => $siteData->toArray()
+                            ];
+                            if (!$preview) {
+                                $this->em->persist($siteData);
+                                $this->em->flush();
+                                $this->loggerService->log(Log::SITE_EDIT, [
+                                    'id' => $primaryId,
+                                    'old' => $existing,
+                                    'new' => $siteData
+                                ]);
+                            }
+                        }
+                        unset($deleted[array_search($siteId, $deleted)]);
+                    } else {
+                        $created[] = $siteData->toArray();
+                        if (!$preview) {
+                            $this->em->persist($siteData);
+                            $this->em->flush();
+                            $this->loggerService->log(Log::SITE_ADD, $siteData->getId());
+                        }
+                    }
+                }
+            }
+        }
+        $deleted = array_values($deleted);
+        if (!$preview) {
+            if ($sitesCount === 0) {
+                throw new \Exception('No sites found');
+            }
+            foreach ($deleted as $siteId) {
+                $site = $existingSites[$siteId];
+                $site->setDeleted(1);
+                $this->em->persist($siteData);
+                $this->em->flush();
+                $this->loggerService->log(Log::SITE_DELETE, $existingSites[$siteId]->getId());
+            }
+        }
+
+        return [
+            'created' => $created,
+            'modified' => $modified,
+            'deleted' => array_values($deleted)
+        ];
+    }
+
+    public function syncAwardees()
+    {
+        $entries = $this->getAwardeeEntriesFromRdr();
+        $awardeesMap = [];
+        foreach ($entries as $entry) {
+            $awardee = $entry->resource;
+            if (empty($awardee->id) || empty($awardee->displayName)) {
+                continue;
+            }
+            if ($awardee->id === 'UNSET') {
+                continue;
+            }
+            $awardeesMap[trim($awardee->id)] = $awardee->displayName;
+        }
+        if (empty($awardeesMap)) {
+            throw new \Exception('No awardees found');
+        }
+        $cmd = $this->em->getClassMetadata(Awardee::class);
+        $connection = $this->em->getConnection();
+        $connection->beginTransaction();
+        try {
+            $connection->query('DELETE FROM ' . $cmd->getTableName());
```

Comment:
Can you use the query builder or DQL to perform this delete instead of using raw SQL?